console/src/view/task.rs

```diff
@@ -123,13 +167,128 @@ impl TaskView {
+            .map(|d| d.make_chart_data(sparkline_area.width - 3))
+            .unwrap_or_default();
+
+        let histogram_sparkline = MiniHistogram::default()
+            .block(
+                Block::default()
+                    .title("Poll Times Histogram")
+                    .borders(Borders::ALL),
+            )
+            .data(&chart_data)
+            .metadata(metadata)
+            .duration_precision(2);
+
         let task_widget = Paragraph::new(metrics).block(block_for("Task"));
         let wakers_widget = Paragraph::new(vec![wakers, wakeups]).block(block_for("Waker"));
         let fields_widget = Paragraph::new(fields).block(block_for("Fields"));
 
+        let percentiles_widget = block_for("Poll Times Percentiles");
+        let (percentiles_1, percentiles_2) = details
+            .map(|d| d.make_percentiles_widgets(5))
+            .unwrap_or_default();
+        let percentiles_widget_1 = Paragraph::new(percentiles_1);
+        let percentiles_widget_2 = Paragraph::new(percentiles_2);
+
         frame.render_widget(Block::default().title(controls), controls_area);
         frame.render_widget(task_widget, stats_area[0]);
         frame.render_widget(wakers_widget, stats_area[1]);
         frame.render_widget(fields_widget, fields_area);
+        frame.render_widget(percentiles_widget, histogram_area[0]);
+        frame.render_widget(percentiles_widget_1, percentiles_columns[0]);
+        frame.render_widget(percentiles_widget_2, percentiles_columns[1]);
+        frame.render_widget(histogram_sparkline, sparkline_area);
+    }
+}
+
+impl Details {
+    /// From the histogram, build a visual representation by trying to make as
+    // many buckets as the width of the render area.
+    fn make_chart_data(&self, width: u16) -> (Vec<u64>, HistogramMetadata) {
+        self.poll_times_histogram()
+            .map(|histogram| {
+                let step_size =
+                    ((histogram.max() - histogram.min()) as f64 / width as f64).ceil() as u64 + 1;
+                // `iter_linear` panics if step_size is 0
+                let data = if step_size > 0 {
+                    let mut found_first_nonzero = false;
+                    let data: Vec<u64> = histogram
+                        .iter_linear(step_size)
+                        .map(|it| it.count_since_last_iteration())
+                        .filter(|count| {
+                            // Remove the 0s from the leading side of the buckets.
+                            // Because HdrHistogram can return empty buckets depending
+                            // on its internal state, as it approximates values.
+                            if *count == 0 && !found_first_nonzero {
+                                false
+                            } else {
+                                found_first_nonzero = true;
+                                true
+                            }
+                        })
+                        .collect();
```

Comment:
nit/take it or leave it: could the `map` and `filter` be combined with `filter_map`? not a big deal either way.